Camera timeout

gie3d opened the pull request:
Base on #123, I created a cameraTimeout props in order to control camera display state. 

Use this to configure how long it should take before the `QRCodeScanner` should be displayed. After that the camera will be inactive and press the view to reactivate it. The `0` default means it's always on.

in `_setCamera`, I tried to follow the implementation of `_setScanning` and believe that whenever the camera is active, it should be ready to scan. That's why I set scanning state in `_setCamera`

gie3d commented:
I updated the conflict in README

renatobenks commented on README.md:
```diff
@@ -166,6 +166,14 @@ default: `0`
+default: `0`
+
+Use this to configure how long it should take before the `QRCodeScanner` should be displayed. After that the camera will be inactive and press the view to reactivate it.
+The `0` default means it's always on.
```
maybe would be great to describe that the number expected in this property must to be in milliseconds, what do you think?

gie3d replied:
Yes that would be good. I'll resubmit the PR

renatobenks commented on index.js:
```diff
@@ -222,8 +233,27 @@ export default class QRCodeScanner extends Component {
+      return (
+        <TouchableWithoutFeedback onPress={() => this._setCamera(true)}>
+          <View style={[styles.camera, styles.inactiveContainer]}>
+            <Text style={styles.inactiveLabel}>Tap to activate camera</Text>
```
wouldn't it be better to have this component to the timeout message coming from the outside instead as a prop?

gie3d replied:
Yes that would be good. I'll resubmit the PR

renatobenks commented:
what else is missing @gie3d? take a look into my review

gie3d commented:
Oh it's been a while since my PR submit. Please let me check. Giving my current schedule, I'll try to get this done within this weekend.

renatobenks commented:
great @gie3d, let me know if you need some help!

gie3d commented:
I tried to test it using react-native-qrcode-scanner-demo but it seems doesn't work at the moment (I still cannot build the demo). So in order to to test this, I asked a question here: https://github.com/moaazsidat/react-native-qrcode-scanner-demo/issues/1 Would it be great to upgrade the demo package to the latest react-native version? so it will be easier to build and this package and other PRs (in my opinion)

renatobenks commented:
@gie3d I see your point here, but I think that you got it badly, because this repository wasn't created to validate PRs or reproduce issues from `react-native-qrcode-scanner`. This is just a first example repository, created just to share how the API should be used in the first release. Probably, just to avoid the whole work on writing a good README in the first releases. So, it was never longer maintained since then, as you can see here https://github.com/moaazsidat/react-native-qrcode-scanner-demo/blob/master/package.json#L23.

Anyway, I think it would be so much better, and ideal, to have an own repository instead, just to validate this PR to finally, move forward with this, which by the way, I would appreciate it if you could share with us this another repository to help you on validating this as well.

gie3d commented:
Thanks for the clarify. Sorry for taking it longer than expected. I pushed the latest update
- added cameraTimeoutView to custom the display when the camera is inactive
- added more detail in README
- deal with fadeIn props as it had an impact during the camera reactivate

renatobenks commented on index.js:
```diff
@@ -3,325 +3,344 @@
+					alignItems: 'center',
+					justifyContent: 'center',
+					backgroundColor: 'transparent',
+					height: Dimensions.get('window').width,
```
is this right as `Dimensions.get('window').width,`? I mean, it shouldn't be like:

```suggestion
					height: Dimensions.get('window').height,
```

gie3d replied:
here I tried to copy the styles.camera because the camera timeout will replace the Camera element. In styles.camera it uses Dimensions.get('window').width as its height. So I followed it.

```js
camera: {
		flex: 0,
		alignItems: 'center',
		justifyContent: 'center',
		backgroundColor: 'transparent',
		height: Dimensions.get('window').width,
		width: Dimensions.get('window').width,
	}
```

renatobenks replied:
fair enough, I've tested and it seems to be ok

renatobenks commented on index.js:
```diff
@@ -3,325 +3,344 @@
+					flex: 0,
+					alignItems: 'center',
+					justifyContent: 'center',
+					backgroundColor: 'transparent',
```
it sounds to be duplicated by the following line https://github.com/moaazsidat/react-native-qrcode-scanner/blob/d5fdd4b8cc9744022d026bd55eabaa829bcfc37c/index.js#L102

as the background `black` seems to be right one:

```suggestion
```

gie3d commented on README.md:
```diff
@@ -184,6 +184,20 @@ default: `0`
+
+propType: `element`
+
+Pass a RN element/component to use it when the camera is inactive (after `cameraTimeout` milliseconds).
```
If the `cameraTimeoutView` is specified without `cameraTimeout`, the `cameraTimeoutView` is not going to be displayed in any circumstance. It will display only if you specify both props and after the camera has been activated for more than `cameraTimeout` milliseconds.

gie3d replied:
Many thanks to your review :) Should I put it like this

#### `cameraTimeoutView`

propType: `element`

Pass a RN element/component to show it when the camera is inactive for `cameraTimeout`  (another props) milliseconds. If the `camearTimeout` is `0` or not specified, this prop will never be used.

renatobenks replied:
> Pass a RN element/component to show it when the camera is inactive for cameraTimeout (another props) milliseconds. If the camearTimeout is 0 or not specified, this prop will never be used.

did you update the docs with this description 👆? it sounds pretty good to me 💯 

by the way, just fix the words typos 😆

gie3d replied:
Ok, I've pushed it. 

> by the way, just make sure that you will fix the words typos above before 😆

I seem don't see the typos you mentioned here 😟, would you mind tell me what went wrong.

renatobenks replied:
> I seem don't see the typos you mentioned here 😟, would you mind tell me what went wrong.

sure, no worries! I'll be put them as commit suggestion in review, sounds quickly to be solved, so closing this conversation

gie3d commented on index.d.ts:
```diff
@@ -66,6 +66,7 @@ export interface RNQRCodeScannerProps {
   buttonPositive?: string;
   checkAndroid6Permissions?: boolean;
   cameraProps?: RNCameraProps;
+  cameraTimeoutView?: JSX.Element;
```
you're right.. I submit a new commit for this.

renatobenks commented on README.md:
```diff
@@ -184,6 +184,20 @@ default: `0`
+
+propType: `element`
+
+Pass a RN element/component to show it when the camera is inactive for `cameraTimeout` (another props) milliseconds. If the `camearTimeout` is 0 or not specified, this prop will never be used.
```
```suggestion
Pass an RN element/component to show it when the camera is inactive for `cameraTimeout` (another prop) milliseconds. If the `cameraTimeout` is 0 or not specified, this prop will never be used.
```
